xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -53,62 +55,38 @@ CollTfCollLenFeature::get_values() const
 
     vector<double> values;
     double value = 0;
-    double coll_len;
-    auto coll_len_iterator = collection_length.find("title");
-    if (coll_len_iterator != collection_length.end())
-	coll_len = (double)coll_len_iterator->second;
-    else
-	coll_len = 0;
+    double coll_len = internal->get_collection_length("title");
 
+    Xapian::Query feature_query = internal->get_query();
     for (Xapian::TermIterator qt = feature_query.get_unique_terms_begin();
 	 qt != feature_query.get_terms_end(); ++qt) {
 	if (is_title_term((*qt))) {
-	    auto coll_tf_iterator = collection_termfreq.find(*qt);
-	    if (coll_tf_iterator != collection_termfreq.end()) {
-		value += log10(1 + (coll_len /
-				    (double)(1 + coll_tf_iterator->second)));
-	    } else {
-		value += log10(1 + coll_len);
-	    }
+	    double tf = internal->get_collection_termfreq(*qt);
+	    value += log10(1 + (coll_len /
+				(double)(1 + tf)));
```

Comment:
The cast to `double` here isn't needed now (since `tf` is a `double` already).  Similarly for some cases below.

Reply:
fixed in a fixup commit.